tests/integration-tests/tests/update/test_update.py

```diff
@@ -749,3 +761,295 @@ def _test_update_queue_strategy_with_running_job(
     _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue1")
     _check_queue_ami(cluster, ec2, pcluster_copy_ami_id, "queue2")
     assert_compute_node_states(scheduler_commands, queue1_nodes, "idle")
+
+
+@pytest.mark.usefixtures("instance")
+def test_dynamic_file_systems_update(
+    region,
+    os,
+    pcluster_config_reader,
+    ami_copy,
+    clusters_factory,
+    scheduler_commands_factory,
+    request,
+    snapshots_factory,
+    efs_stack_factory,
+    vpc_stack,
+    key_name,
+    s3_bucket_factory,
+    test_datadir,
+    fsx_factory,
+    svm_factory,
+    open_zfs_volume_factory,
+):
+    """Test update shared storages."""
+    existing_ebs_mount_dir = "/existing_ebs_mount_dir"
+    existing_efs_mount_dir = "/existing_efs_mount_dir"
+    fsx_lustre_mount_dir = "/fsx_lustre_mount_dir"
+    fsx_ontap_mount_dir = "/fsx_ontap_mount_dir"
+    fsx_open_zfs_mount_dir = "/fsx_open_zfs_mount_dir"
+    new_ebs_mount_dir = "/new_ebs_mount_dir"
+    new_raid_mount_dir = "/raid_dir"
+    new_efs_mount_dir = "/new_efs_mount_dir"
+    new_lustre_mount_dir = "/new_lustre_mount_dir"
+    ebs_mount_dirs = [new_ebs_mount_dir, existing_ebs_mount_dir]
+    efs_mount_dirs = [existing_efs_mount_dir, new_efs_mount_dir]
+    fsx_mount_dirs = [new_lustre_mount_dir, fsx_lustre_mount_dir, fsx_open_zfs_mount_dir, fsx_ontap_mount_dir]
+    all_mount_dirs = ebs_mount_dirs + [new_raid_mount_dir] + efs_mount_dirs + fsx_mount_dirs
+
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "s3_test_file"), "s3_test_file")
```

Comment:
My understanding is that we upload this file to a bucket attached to the FSx Lustre file system.
Why do we need that? Isn't enough to mount a FSx Lustre file system without an underlying bucket?

Reply:
The bucket is used in the integration test to test fsx functionality after update cluster with fsx to test data repository task and export path

Reply:
Thanks for the explanation. I agree with you that the more we test the better it is, but I'd like to understand this point a bit better. The goal of the test here is to verify that file systems are correctly mounted/unmounted in a way that is ruled by the QueueUpdateStrategy. Isn't enough to check the mount points instead of using S3?

Reply:
Previously, we don't support update cluster to add filesystems. As we support it now, I just want to verify that when we update a cluster to add fsx, everything of fsx works as expected. I agree it is a nice to have, I don't have a strong opinion on keeping it or removing it. Both are fine to me.